File: back-end/src/utils/responseHelper.js
/**
 * Utilitários para padronizar respostas da API
 */
const responseHelper = {
  /**
   * Resposta de sucesso
   */
  success(res, data, message = 'Operação realizada com sucesso', statusCode = 200) {
    return res.status(statusCode).json({
      status: 'success',
      message,
      data,
    });
  },

  /**
   * Resposta de erro
   */
  error(res, message = 'Erro ao processar requisição', statusCode = 400) {
    return res.status(statusCode).json({
      status: 'error',
      message,
    });
  },

  /**
   * Resposta de criação
   */
  created(res, data, message = 'Registro criado com sucesso') {
    return this.success(res, data, message, 201);
  },

  /**
   * Resposta não encontrado
   */
  notFound(res, message = 'Registro não encontrado') {
    return this.error(res, message, 404);
  },

  /**
   * Resposta de validação
   */
  validationError(res, errors, message = 'Erro de validação') {
    return res.status(400).json({
      status: 'error',
      message,
      errors,
    });
  },
};

module.exports = responseHelper;

File: backend/api/utils/response_helper.py
from rest_framework.response import Response


def success(data=None, message: str = "Operação realizada com sucesso", status_code: int = 200):
    return Response(
        {
            "status": "success",
            "message": message,
            "data": data,
        },
        status=status_code,
    )


def created(data=None, message: str = "Registro criado com sucesso"):
    return success(data, message, status_code=201)


def error(message: str = "Erro ao processar requisição", status_code: int = 400):
    return Response(
        {
            "status": "error",
            "message": message,
        },
        status=status_code,
    )


